tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -777,6 +778,27 @@ func (az *AzureProvider) CreateWindowsVM() (*types.Credentials, error) {
 		log.Printf("failed to get the IP address of %s: %s", az.IpName, ipErr)
 		*ipAddress = az.IpName
 	}
+
+	// Build new credentials structure to be used by other actors. The actor is responsible for checking if
+	// the credentials are being generated properly. This method won't guarantee the existence of credentials
+	// if the VM is spun up
+	credentials := types.NewCredentials(instanceName, *ipAddress, adminPassword, winUser)
+	w.Credentials = credentials
+
+	// Setup Winrm and SSH client so that we can interact with the Windows Object we created
+	if err := w.SetupWinRMClient(); err != nil {
+		return nil, fmt.Errorf("failed to setup winRM client for the Windows VM: %v", err)
+	}
+
+	// Wait for some time before starting configuring of ssh server. This is to let sshd service be available
+	// in the list of services
+	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
+	// has those services present
+	time.Sleep(time.Minute)
+	if err := w.GetSSHClient(); err != nil {
+		return w, fmt.Errorf("failed to get ssh client for the Windows VM created: %v", err)
+	}
+
```

Comment:
Same comment here and if possible make it a common helper.

Reply:
Added a TODO. https://github.com/openshift/windows-machine-config-bootstrapper/pull/155/files#diff-356b22f90392ac8fc81c91247ebede89R205